### PR TITLE
Add accumulate exercise

### PR DESCRIPTION
**Pull request by lpil**

### CODE REVIEW

File: exercises/accumulate/test/Main.purs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import Control.Monad.Eff (Eff)
+import Test.Unit (suite, test)
+import Test.Unit.Main (runTest)
+import Test.Unit.Assert as Assert
+import Data.List (List, fromFoldable)
+import Data.String as String
+import Accumulate (accumulate)
+
+
+main :: Eff _ Unit
```
Comment (paf31) [inline exercises/accumulate/test/Main.purs]: I think this is fine. The IDE should only give a warning.

Comment (lpil) [inline exercises/accumulate/test/Main.purs]: Thanks. I'll have see if I can make vim ignore this warning.

**Comment by lpil** [comment]
Happy to merge if you are :)

**Comment by paf31** [comment]
Oops, I added comments but forgot to submit them, sorry.

File: exercises/accumulate/examples/src/Accumulate.purs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Accumulate
+  ( accumulate
+  ) where
+
+import Data.List (List(Nil), (:), reverse)
+
+
+accumulate :: forall a b . (a -> b) -> List a -> List b
+accumulate f values = accumulate' f Nil values
```
Comment (lpil) [inline exercises/accumulate/examples/src/Accumulate.purs]: I just find TCO easier to read and write these days. I don't think that exercism actually exposes these example solutions to students anywhere so I'm not really sure what we're aiming for here. Should it be trying to be beginner friendly, or should it try to be a more robust solution?

File: exercises/accumulate/examples/src/Accumulate.purs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+
+accumulate' :: forall a b . (a -> b) -> List b -> List a -> List b
```
Comment (lpil) [inline exercises/accumulate/examples/src/Accumulate.purs]: Sounds good.

Is there a convention about when to use `where` and when to use `let`? I've been using `let` as I find it a bit easier to read with only one style.

File: exercises/accumulate/test/Main.purs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+main = runTest do
+  suite "Pangram.isPangram" do
```
Comment (lpil) [inline exercises/accumulate/test/Main.purs]: Oops. Copy/paste mistake.

File: exercises/accumulate/bower.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  "name": "hello-world",
+  "ignore": [
+    "**/.*",
+    "node_modules",
+    "bower_components",
+    "output"
+  ],
+  "dependencies": {
+    "purescript-prelude": "^2.1.0",
+    "purescript-sets": "^2.0.0"
```
Comment (lpil) [inline exercises/accumulate/bower.json]: Yet more sloppy copy pasting :)

**Comment by lpil** [comment]
Updated the branch @paf31

**Comment by paf31** [comment]
Looks great, thanks! Could you please fix the merge conflicts?

File: exercises/accumulate/examples/src/Accumulate.purs (hunk 3 above)
Comment (paf31) [inline exercises/accumulate/examples/src/Accumulate.purs]: I don't have a strong preference, but sometimes `let` can be useful if you want to place a let between two function arguments, which you can't do with `where`.

File: exercises/accumulate/examples/src/Accumulate.purs (hunk 2 above)
Comment (paf31) [inline exercises/accumulate/examples/src/Accumulate.purs]: > I don't think that exercism actually exposes these example solutions to students anywhere

Ah, I didn't realize that 😄  I think this level is fine in that case.

**Comment by lpil** [comment]
I've rebased on master :)

**Pull request merged**